### ISSUE TITLE
[BUG]? Can't get repeater translation inside templatefields working

### ISSUE THREAD

**Issue by danielschleindlsperger** [AnimalDesign/bolt-translate#57]
First of all thank you so much for your great work on bolt and also for the great support in the slack channel! 

Now to my problem:
I have a section content type which only has a templateselect. For my different types of sections I have different template and templatefields. Some of them contain repeaters, which I can't get to work with the translate plugin.

 - Translate Version: 3.1.1
 - Install type: Extension Repository
 - Bolt Version: 3.0.11
 - PHP version: 7.0
 - Used webserver: Apache version 2.4.23

### config
``` yaml
locales:
    de_DE:
        label: Deutsch
        slug: de
    en_GB:
        label: English
        slug: en
```

### sections contenttype:
``` yaml
sections:
    name: Sections
    singular_name: Section
    fields:
        locale:
            type: locale
            group: content
        title:
            type: text
            class: large
            group: content
        slug:
            type: slug
            uses: title
        template:
            type: templateselect
            filter: '*_section.twig'
            is_translateable: true
            group: content
        slug:
            type: slug
            uses: title
    record_template: image-and-text_section.twig
    default_status: published
    viewless: true
    searchable: false
```

### templatefields for section:
``` yaml

templatefields:
    sections\image-and-text_section.twig:
        headline:
            type: text
            class: large
        content:
            type: html
            height: 200px
        image:
            type: image
            attrib: title
            extensions: [ gif, jpg, png ]
        contentlink:
            type: text
            label: Link
        contentlink_text:
            type: text
            label: Link Text
    sections\was-uns-auszeichnet_section.twig:
        headline:
            type: text
            title: Headline
        content_matrix:
            type: repeater
            limit: 10
            fields:
                title:
                    type: text
                    label: Titel
                bubble_size:
                    type: text
                    label: Size (small, medium, big)
                content_text:
                    type: text
                    label: Text
                bubble_color:
                    label: Color
                    type: select
                    values: { 'red': "Red", 'medium-gray': "Medium Gray", 'dark-gray': "Dark Gray", 'blue': "Blue" }
    sections\stage_section.twig:
        slides:
            type: repeater
            fields:
                image:
                    type: image
                    attrib: [title, alt]
                    extensions: [ gif, jpg, png ]
                    label: Image
                teaser_text:
                    type: html
                    height: 300px
                    label: Teaser Text
                overlay_text_top_small:
                    type: text
                    label: Image Overlay Text Top Small
                overlay_text_center_big:
                    type: text
                    label: Image Overlay Text Center Big
                overlay_text_bottom_small:
                    type: text
                    label: Image Overlay Text Bottom Small
                contentlink:
                    type: text
                    label: CTA Link
                contentlink_text:
                    type: text
                    label: CTA Text
```

The section without repeater works fine, in the other two I can either save my primary language (de) or if I try to save content to my secondary language (en) it will save neither.

**Comment by SahAssar** [comment]
Hey, I almost got this working, but you should know that there seems to be another bug that you run into with that repeater, see bolt/bolt#5652

I'll post an update here when I got it fixed :)

**Comment by cigear** [comment]
There is problem as below.
private function registerOverrides(Application $app) in  TranslateExtension.php
-----------------------------------------------------------
  $app['schema.content_tables'] = $app->extend(
            'schema.content_tables',
            function ($contentTables) use ($app) {
                $config = $app['translate.config'];
                $platform = $app['db']->getDatabasePlatform();
                $prefix = $app['schema.prefix'];
                $contentTypes = $app['config']->get('contenttypes');
      
                foreach (array_keys($contentTypes) as $contentType) {
                    $contentTables[$contentType] = $app->share(function () use ($platform, $prefix, $config) {
                        return new Storage\ContentTypeTable($platform, $prefix, $config);
                    });
                }

                return $contentTables;
            }
        );
--------------------------------------------

 $contentTables[$contentType] must be  $contentTables[$tablename];

You could get   $tableName as below.
  $tableName = $contentTypes[$contentType]['tablename'];

**Comment by cigear** [comment]
add more comment. If you define some content types as en-pages or en_pages including _ , - , two tables
as en-pages, en_pages will be created and an error will occur.

**Comment by SahAssar** [comment]
@cigear These issues do not seem related to the actual issue this thread is about, could you post them in a separate issue?

**Comment by SahAssar** [comment]
Btw, I have a fix for this incoming, I was waiting for bolt/bolt#5670 to get ready so we knew how repeaters in templatefields would behave on the frontend before committing. 

The fix should land tomorrow, or this weekend at the latest.

**Comment by SahAssar** [comment]
Fixed in #64 which will be useable in the next stable bolt release.
